backend/data-store/db-client/mail.go

```diff
@@ -0,0 +1,265 @@
+			}); err != nil {
+				st.Log.Error("error incrementing email tracker", zap.Error(err))
+				continue
+			}
+
+			emailsToSend = append(emailsToSend, emailDetails)
+			continue
+		}
+	}
+
+	return &ds.EmailResponse{
+		Emails: emailsToSend,
+	}, nil
+}
+
+func (st *Store) getEmailsForIndividualRatings(ctx context.Context) ([]*EmailRow, error) {
+	res, err := queries.New(st.Pool).GetIndividualRatingsEmails(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting emails for individual ratings: %w", err)
+	}
+
+	return lo.Map(res, func(r queries.GetIndividualRatingsEmailsRow, _ int) *EmailRow {
+		return &EmailRow{
+			AssessorID:     r.AssessorID,
+			SubmissionID:   r.SubmissionID,
+			SubmissionName: r.SubmissionName,
+			Year:           r.Year,
+			SetBeforeDate:  r.SetBeforeDate,
+			IsCreated:      r.IsCreated,
+			FirstName:      r.FirstName,
+			LastName:       r.LastName,
+			Email:          r.Email,
+			RatingType:     ds.RatingType_INDIVIDUAL,
+			RatingSqlType:  queries.ProjectRatingTypeIndividual,
+		}
+	}), nil
+}
+
+func (st *Store) getEmailsForFinalRatings(ctx context.Context) ([]*EmailRow, error) {
+	res, err := queries.New(st.Pool).GetFinalRatingsEmails(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting emails for final ratings: %w", err)
+	}
+
+	return lo.Map(res, func(r queries.GetFinalRatingsEmailsRow, _ int) *EmailRow {
+		return &EmailRow{
+			AssessorID:     r.AssessorID,
+			SubmissionID:   r.SubmissionID,
+			SubmissionName: r.SubmissionName,
+			Year:           r.Year,
+			SetBeforeDate:  r.SetBeforeDate,
+			IsCreated:      r.IsCreated,
+			FirstName:      r.FirstName,
+			LastName:       r.LastName,
+			Email:          r.Email,
+			RatingType:     ds.RatingType_FINAL,
+			RatingSqlType:  queries.ProjectRatingTypeFinal,
+		}
+	}), nil
+}
+
+func (st *Store) getEmailsForInitialRatings(ctx context.Context) ([]*EmailRow, error) {
+	res, err := queries.New(st.Pool).GetInitialRatingsEmails(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting emails for initial ratings: %w", err)
+	}
+
+	return lo.Map(res, func(r queries.GetInitialRatingsEmailsRow, _ int) *EmailRow {
+		return &EmailRow{
+			AssessorID:     r.AssessorID,
+			SubmissionID:   r.SubmissionID,
+			SubmissionName: r.SubmissionName,
+			Year:           r.Year,
+			SetBeforeDate:  r.SetBeforeDate,
+			IsCreated:      r.IsCreated,
+			FirstName:      r.FirstName,
+			LastName:       r.LastName,
+			Email:          r.Email,
+			RatingType:     ds.RatingType_INITIAL,
+			RatingSqlType:  queries.ProjectRatingTypeInitial,
+		}
+	}), nil
+}
+
+func getWithBackoff[T any](
+	ctx context.Context,
+	logger *zap.Logger,
+	backoff int,
+	getter func(context.Context) (T, error),
+) (*T, error) {
+
+	for i := 0; i < backoff; i++ {
+		emails, err := getter(ctx)
+		if err != nil {
+			logger.Error("error getting emails", zap.Int("interatio", i), zap.Error(err))
+			continue
+		}
+		return &emails, nil
+	}
+	return nil, errors.New("error getting emails")
+}
```

Comment:
Just out of curiosity - did you encounter any throttling? Or it's just preventional?

Reply:
No. This is just a safety measure for not missing emails if any transient errors arise. For other operations, if they fail user can re initiate them - here since it's cron job we would need to wait 24h. We wouldn't want our emails to be day late just because of some temporary connection loss now, would we?